### PR TITLE
Bluetooth: audio: ascs: Move control point pointer out of bt_ascs

### PR DESCRIPTION
**Pull request by MariuszSkamra**
bt_ascs is per connection, so the control point attribute pointer should not be part of this structure. This moves the attribute pointer out of the bt_ascs and adds getter function that lazy inits the pointer.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

### CODE REVIEW

File: subsys/bluetooth/audio/ascs.c [hunk 1]
```diff
@@ -2451,7 +2456,7 @@ static ssize_t ascs_cp_write(struct bt_conn *conn,
 	}
 
 respond:
-	bt_gatt_notify(ascs->conn, ascs->control_point_attr, rsp_buf.data, rsp_buf.len);
+	bt_gatt_notify(ascs->conn, control_point_attr(), rsp_buf.data, rsp_buf.len);
```
Comment (Thalley) [inline subsys/bluetooth/audio/ascs.c:2459]: I would suggest to simply use `bt_gatt_notify_uuid` with `BT_UUID_ASCS_ASE_CP` instead, which seems to do just about the same thing, but at the GATT layer instead. That way we can omit the `control_point_attr` and `control_point_attr_cb` functions

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/audio/ascs.c (hunk 1 above)
Comment (MariuszSkamra) [reply subsys/bluetooth/audio/ascs.c:2459]: Hmm, I see the advantage of using `bt_gatt_notify_uuid` in terms of removing of code, while in terms of optimization  `bt_gatt_notify_uuid` iterates the db every time it's used

File: subsys/bluetooth/audio/ascs.c [hunk 2]
```diff
@@ -2468,13 +2448,18 @@ static ssize_t ascs_cp_write(struct bt_conn *conn,
...
 	BT_GATT_PRIMARY_SERVICE(BT_UUID_ASCS),
-	LISTIFY(CONFIG_BT_ASCS_ASE_SNK_COUNT, BT_ASCS_ASE_SNK_DEFINE, (,)),
-	LISTIFY(CONFIG_BT_ASCS_ASE_SRC_COUNT, BT_ASCS_ASE_SRC_DEFINE, (,)),
 	BT_AUDIO_CHRC(BT_UUID_ASCS_ASE_CP,
 		      BT_GATT_CHRC_WRITE | BT_GATT_CHRC_WRITE_WITHOUT_RESP | BT_GATT_CHRC_NOTIFY,
 		      BT_GATT_PERM_WRITE_ENCRYPT,
 		      NULL, ascs_cp_write, NULL),
 	BT_AUDIO_CCC(ascs_cp_cfg_changed),
+	LISTIFY(CONFIG_BT_ASCS_ASE_SNK_COUNT, BT_ASCS_ASE_SNK_DEFINE, (,)),
+	LISTIFY(CONFIG_BT_ASCS_ASE_SRC_COUNT, BT_ASCS_ASE_SRC_DEFINE, (,)),
```
Comment (Thalley) [inline subsys/bluetooth/audio/ascs.c:2457]: Why were these moved?

File: subsys/bluetooth/audio/ascs.c (hunk 1 above)
Comment (MariuszSkamra) [reply subsys/bluetooth/audio/ascs.c]: I've changed to use bt_gatt_notify_uuid, and moved the CP attribute to be the second entry in attrs array.
Anyway, the `&ascs_svc.attrs[1]` could be used here IMO instead

**Review by Thalley — APPROVED** [review]

File: subsys/bluetooth/audio/ascs.c (hunk 2 above)
Comment (MariuszSkamra) [reply subsys/bluetooth/audio/ascs.c:2457]: To limit the number of iterations to find the attribute with UUID given.

**Review by szymon-czapracki — APPROVED** [review]

**Review by sjanc — APPROVED** [review]

**Review by Thalley — APPROVED** [review]
